File: Algorithm/BOJ/Python/Breadth-First Search/2606.py
# -*- coding: utf-8 -*-
"""
Created on Mon Dec 20 19:57:57 2021

@author: MJH
"""



visited = []
graph = []
cnt = 0


def bfs(edge: int) -> None:
    global visited, graph, cnt
    
    queue = [edge]
    while queue:
        for next_ in graph[queue.pop()]:
            if not visited[next_]:
                visited[next_]=True
                queue.append(next_)
                cnt += 1
  
def main():
    global visited, graph, cnt
    node = int(input())
    edge = int(input())
    
    visited = [False for _ in range(node+1)]
    graph = [[] for _ in range(node+1)]
    
    
    for _ in range(edge):
        u, v = map(int, input().split())
        graph[u].append(v)
        graph[v].append(u)
    
    bfs(1)
    print(cnt-1)


if __name__ == '__main__':
    main()        

File: Algorithm/BOJ/C++/Breadth-First Search/2606.cpp
#include <iostream>
#include <queue>
#include <vector>

using namespace std;

void bfs(int edge);
int cnt = 0;
vector<vector<int>> network;
vector<bool> visited;



int main() {
	int node, edge, u, v;
	cin >> node >> edge;

	network.resize(node + 1);
	visited.resize(node + 1);
	
	for (int i = 0; i < edge; i++) {
		cin >> u >> v;
		network[u].push_back(v);
		network[v].push_back(u);
	}

	bfs(1);
	cout << cnt - 1;
}



void bfs(int edge) {
	int next_;
	queue<int> queue;
	queue.push(edge);

	while (!queue.empty()) {
		next_ = queue.front();
		queue.pop();
		for (int next : network[next_]) {
			if (!visited[next]) {
				visited[next] = true;
				queue.push(next);
				cnt++;
			}
		}
	}

}